Added pre-processor for text-media matching

telescopic opened the pull request:
* Added basic text-media matching pre-processor
* Added numpy to requirements.txt

TODO:
* add tests (will be addressed in a later PR)

tmadhuri commented on classifier_and_summarizer/summarizer/text_media_matching_preprocessor.py:
```diff
@@ -0,0 +1,99 @@
+'''
+from numpy import maximum as matrix_maximum
+from sklearn.metrics.pairwise import cosine_similarity
+
```
Now we have 3 files for text_media_matching. Would it make sense to create a separate directory? Also, TextMediaMatcher.py could be renamed to reflect that it deals with preference matrice's creation.

telescopic replied:
Thanks for pointing this out - created a directory separately for that,
However i feel that leaving it as TextMediaMatcher would be preferable since its main functionality is to return the matched text and media and not the preference matrix creation - which is just a pre-processing step to apply the matching.  Let me know if you think there's a better way to do this - would be great to hear your thoughts on this!

telescopic replied:
Raised an issue to possibly amend this in a later PR!

tmadhuri commented on classifier_and_summarizer/summarizer/text_media_matching_preprocessor.py:
```diff
@@ -0,0 +1,99 @@
+            )
+        )
+
+    def _prune_larger_list(self):
```
Try splitting this method.

telescopic replied:
Updated accordingly!

telescopic commented:
@tmadhuri I've refactored and restructured the code. I've also added some code to take care of what i feel are edge case. Will add tests for this in a future PR. Please do take a look at this and request changes if necessary!

telescopic commented on classifier_and_summarizer/summarizer/text_media_matching/text_media_matching_preprocessor.py:
```diff
@@ -0,0 +1,123 @@
+            )
+        )
+
+    def _initialize_auxiliary_variables(self):
```
Updated accordingly!
